xarray/backends/zarr.py

```diff
@@ -320,6 +320,15 @@ def _validate_existing_dims(var_name, new_var, existing_var, region, append_dim)
         )
 
 
+def _put_attrs(zarr_obj, attrs):
+    for key, value in attrs.items():
+        try:
+            zarr_obj.attrs[key] = value
+        except TypeError as e:
+            raise TypeError(f"Invalid attr {key!r}: {value!r}. {e!s}") from e
+    return zarr_obj
```

Comment:
Yes - I started with an error as close as possible to the one that was already implemented, but I also think that it's better to avoid the loop.

For example, if I assign a DataArray to an attribute, I get the following error from `zarr`:
```
TypeError: Object of type DataArray is not JSON serializable
```
I think it's OK if `xarray` doesn't explicitly say which attributes generate the problem (i.e., avoid the loop), but maybe `xarray` should say that the issue is in the attributes (i.e., keep the try/except to raise a more informative error).

What do you think?